mergify/labels-copier/copy.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# Environment variables:
+#  PR_URL
+#  REPOSITORY_URL
+#  ADDITIONAL_LABELS
+#  EXCLUDED_LABEL
+#  DRY_RUN
+#  RUNNER_DEBUG
+#
+# You can run this locally for testing purposes:
+#
+# $ REPOSITORY_URL=https://github.com/elastic/apm-server PR_URL=https://github.com/elastic/apm-server/pull/15035 EXCLUDED_LABEL=backport-* DRY_RUN=true sh mergify/labels-copier/copy.sh
+#
+
+set -eo pipefail
+
+# Support debugging in GitHub actions when RUNNER_DEBUG is set.
+if [ -n "$RUNNER_DEBUG" ] ; then
+  set -x
+fi
+
+add_label() {
+  local label=$1
+  local pr_url=$2
+
+  if [ "$DRY_RUN" == "true" ]; then
+    echo ">> [dry-run]: $label will be added"
+  else
+    gh pr edit --add-label "$label" "$pr_url"
+  fi
+}
+
+
+pr_number=$(gh pr view --json body -q ".body" "$PR_URL" | sed -n -e '/automatic backport of pull request/,/done/p' | cut -d"#" -f2 | cut -d" " -f1)
```

Comment:
Is there any other way ito avoid parse the body?

Reply:
No, unfortunately, the PR reference in Mergify is within the PR description.

See https://github.com/elastic/apm-server/pull/15186

<img width="646" alt="image" src="https://github.com/user-attachments/assets/2a665742-9592-4969-b2b3-5f3eb494928c" />

Reply:
And also in the PR title

<img width="1133" alt="image" src="https://github.com/user-attachments/assets/7243e6e9-a92c-460d-a45c-6db8d1573cf9" />

That's something configured at https://github.com/elastic/apm-server/blob/ac892161040578cf3b10ab77662cd39d6b5b7c8c/.mergify.yml#L72

Reply:
This code get the label of a PR

```javascript
          const { data } = await github.rest.issues.listLabelsOnIssue({
            issue_number: process.env.issue,
            owner: context.repo.owner,
            repo: context.repo.repo,
          })
```